### PR TITLE
add feature to add ip address exception in lockdown

### LINKED ISSUE #13: Allow lockdown for certain IP addresses
Hello,
I have a such case:
 1. development service (has lockdown enabled)
 2. another development service with works with 1st service

I do not want to expose certain urls for 2nd service, I want to expose bunch of urls - so would be great to have option for exception for IP address

### PR DESCRIPTION
**Pull request by pfk-beta**
I want to resolve #13

### CODE REVIEW

File: lockdown/tests/tests.py [hunk 1]
```diff
@@ -57,6 +57,28 @@ def test_url_exceptions(self):
         response = self.client.get(self.locked_url)
         self.assertEqual(response.content, self.locked_contents)
 
+    @patch('lockdown.tests.tests.middleware.settings.REMOTE_ADDR_EXCEPTIONS',
+           ('192.168.0.1',))
+    def test_remote_addr_exception_lock(self):
+        """Test that a page isn't locked when client IP is in exception list
```
Comment (Dunedan) [inline lockdown/tests/tests.py]: This docstring is the same as for the following test and doesn't seem to be correct.

File: README.rst [hunk 2]
```diff
@@ -164,6 +164,16 @@ locked. For example::
         r'\.json$',   # unlock JSON API
     )
 
+LOCKDOWN_REMOTE_ADDR_EXCEPTIONS
+-------------------------------
+An optional list/tuple of IPv4 addresses to be matched agains incomming
```
Comment (Dunedan) [inline README.rst]: Can you please remove the mention of the tuple and make it a list in the example and the tests? I'd like to keep it in line with Django's `INTERNAL_IPS` which takes a list as well.

Comment (Dunedan) [inline README.rst]: If I'm not mistaken that option should work fine with IPv6-addresses as well, so I suggest to reflect that in the README.

**Review by Dunedan — COMMENTED** [review]
First of all: Thanks a lot for your contribution :+1:

What's you motivation of using a separate configuration option instead of re-using Django's [`INTERNAL_IPS`](https://docs.djangoproject.com/en/2.0/ref/settings/#internal-ips) setting?

Aside from that I have some minor review comments, you'll find inline. I'd be great if you could address them.

File: README.rst (hunk 2 above)
Comment (pfk-beta) [reply README.rst]: After 6 hours of work, I didn't think that it will work with ipv6

File: lockdown/tests/tests.py (hunk 1 above)
Comment (pfk-beta) [reply lockdown/tests/tests.py]: Typo - will be changed.

File: README.rst (hunk 2 above)
Comment (pfk-beta) [reply README.rst]: Ye sure, I can change. But I took example from previous code of tests, e.g. https://github.com/Dunedan/django-lockdown/blob/f659d3a95c8a9e717d2dd80fd075f6b2830eb898/lockdown/tests/tests.py#L50 or https://github.com/Dunedan/django-lockdown/blob/f659d3a95c8a9e717d2dd80fd075f6b2830eb898/lockdown/tests/tests.py#L60 
I want to be consistent with other examples.

**Comment by pfk-beta** [comment]
> What's you motivation of using a separate configuration option instead of re-using Django's INTERNAL_IPS setting?

There was no information in README, that I can use this INTERNAL_IPS. So is my PR useless?

**Comment by Dunedan** [comment]
> There was no information in README, that I can use this INTERNAL_IPS. So is my PR useless?

No, it definitely isn't. `django-lockdown` currently doesn't care about the `INTERNAL_IPS`. I was just wondering if it'd make sense to use `INTERNAL_IPS` for your PR instead of introducing `LOCKDOWN_REMOTE_ADDR_EXCEPTIONS`.

File: README.rst (hunk 2 above)
Comment (Dunedan) [reply README.rst]: Yes, the other parameters often promote the ability to use tuples or lists. Going forward I prefer keeping it clean and similar to how Django itself does it and take lists for new parameters. Old parameters of course will keep tuples and lists to keep backward-compatibility.

Comment (Dunedan) [reply README.rst]: Shouldn't it just work? In the end whether IPv4 or IPv6 is used `REMOTE_ADDR` contains a string and that string is matched against the entries in `LOCKDOWN_REMOTE_ADDR_EXCEPTIONS`, isn't it?

**Comment by pfk-beta** [comment]
image

For me INTERNAL_IPS means something else than IP_EXCEPTIONS. I've attached image - correct me if I'm wrong. Maybe we can merge INTERNAL_IPS with IP_EXCEPTIONS, to use in final locking mechanism?

**Comment by Dunedan** [comment]
> For me INTERNAL_IPS means something else than IP_EXCEPTIONS. I've attached image - correct me if I'm wrong.

All right, I just wanted to ensure that your use case is a different one than for `INTERNAL_IPS`, otherwise re-using the values from the `INTERNAL_IPS` option would have been the better approach. :+1:

**Comment by Dunedan** [comment]
Thanks for the adjustments. :+1:
The only concern I still have is the missing IPv6-support. I'll try to reproduce the problems you experienced. Maybe we can figure out what's the culprit there.

**Comment by Dunedan** [comment]
Finally I managed to test it with IPv6 and it works just fine. :+1:
Merging this PR now. Sorry for the long wait.

**Comment by pfk-beta** [comment]
good job

**Pull request merged**